### PR TITLE
feat(Workspaces): allow admin to invite external users

### PR DESCRIPTION
**Pull request by cheikhgwane**
Allow workspace admin to invite external users.

## Changes

Please list / describe the changes in the codebase for the reviewer(s).

- Add  a workspace sign up page

## How/what to test

A few words about what needs to be tested, along with specific testing instructions if needed.

## Screenshots / screencast

image

### CODE REVIEW

**Review by qgerome — COMMENTED** [review]
The flow is great ! 
My remaining remarks are:
- Do we log in the newly created user automatically or not ?
- We should display the list of invitations in the settings

**Comment by qgerome** [comment]
> The flow is great ! My remaining remarks are:
> 
> * Do we log in the newly created user automatically or not ?
> * We should display the list of invitations in the settings

@pvanliefland that's for you as well in fact

**Review by qgerome — CHANGES_REQUESTED** [review]

File: src/pages/workspaces/[workspaceSlug]/signup.tsx [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+import Page from "core/components/Page";
+
+import Field from "core/components/forms/Field";
+import clsx from "clsx";
+import { useJoinWorkspaceMutation } from "workspaces/graphql/mutations.generated";
+import { JoinWorkspaceError } from "graphql-types";
+import { useTranslation } from "react-i18next";
+
+type Props = {
+  page: number;
```
[the remark this answers is not in the document]
Comment (cheikhgwane) [reply src/pages/workspaces/[workspaceSlug]/signup.tsx:19]: Not needed, unintentional.

**Comment by pvanliefland** [comment]
> > The flow is great ! My remaining remarks are:
> > 
> > * Do we log in the newly created user automatically or not ?
> > * We should display the list of invitations in the settings
> 
> @pvanliefland that's for you as well in fact

Logging in the newly created user is not necessary IMHO.

I agree, displaying the list of invitations in the settings is a good idea. It might mitigate the need for an alert / notification after sending the invitation, let's discuss it.

**Comment by cheikhgwane** [comment]
[Screencast from 11.07.2023 11:40:24.webm](https://github.com/BLSQ/openhexa-frontend/assets/25453621/e92e9a0b-169b-47d6-9739-7f7ff7eb42f3)

@pvanliefland  @qgerome

**Review by qgerome — APPROVED** [review]

**Pull request merged**